### PR TITLE
LPD-42640 Bumping to graphql-java@19.11 to address security issue

### PR DESCRIPTION
**Pull request by GaborKomaromi**

### CODE REVIEW

**Comment by GaborKomaromi** [comment]
ci:test:sf

**Comment by GaborKomaromi** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
@albertojml could you review (and test) this PR?

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
Hey @GaborKomaromi ,

I have executed the `GraphQLServletTest` of the Vulcan module and see the results:

---

Before the PR

```
com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest STANDARD_OUT
    Loading jar:file:/home/carlos/Liferay/Proyecto/bundles/tomcat-9.0.90/webapps/ROOT/WEB-INF/shielded-container-lib/portal-impl.jar!/system.properties

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutation STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutation PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutationWithGraphQLNamespace STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutationWithGraphQLNamespace PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQuery STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQuery PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryDepthLimit STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryDepthLimit PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryPagination STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryPagination PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryWithGraphQLNamespace STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryWithGraphQLNamespace PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testSchema STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testSchema PASSED

> Task :apps:portal-vulcan:portal-vulcan-test:stopTestableTomcat


BUILD SUCCESSFUL in 1m 13s
88 actionable tasks: 4 executed, 84 up-to-date
```

---

*After the PR*

```
> Task :apps:portal-vulcan:portal-vulcan-test:testIntegration

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest STANDARD_OUT
    Loading jar:file:/home/carlos/Liferay/Proyecto/bundles/tomcat-9.0.90/webapps/ROOT/WEB-INF/shielded-container-lib/portal-impl.jar!/system.properties

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutation STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutation PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutationWithGraphQLNamespace STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testMutationWithGraphQLNamespace PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQuery STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQuery PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryDepthLimit STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryDepthLimit PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryPagination STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryPagination PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryWithGraphQLNamespace STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testQueryWithGraphQLNamespace PASSED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testSchema STARTED

com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest > testSchema FAILED
    java.lang.NullPointerException
    	at com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest._getJSONObject(GraphQLServletTest.java:843)
    	at com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest._assertGraphQLSchemaField(GraphQLServletTest.java:817)
    	at com.liferay.portal.vulcan.internal.graphql.servlet.test.GraphQLServletTest.testSchema(GraphQLServletTest.java:456)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
    	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:228)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:307)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:86)
    	at java.lang.Thread.run(Thread.java:750)

7 tests completed, 1 failed
There were failing tests. See the report at: file:///home/carlos/Liferay/Proyecto/liferay-portal/modules/apps/portal-vulcan/portal-vulcan-test/build/reports/tests/testIntegration/index.html

> Task :apps:portal-vulcan:portal-vulcan-test:stopTestableTomcat


BUILD SUCCESSFUL in 1m 4s
88 actionable tasks: 4 executed, 84 up-to-date
```

Could you please check why?

**Comment by ccorreagg** [comment]
@GaborKomaromi Can you please update the configurations to run the relevant tests if there is a change in the build.gradle file?

**Comment by GaborKomaromi** [comment]
@ccorreagg I made the requested changes, the failing test is now passing.

File: modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java [hunk 1]
```diff
@@ -458,48 +453,25 @@ public void testSchema() throws Exception {
 		_assertGraphQLSchemaField(
 			false, mutationFieldsJSONArray, true, "testPath_v1_0");
 		_assertGraphQLSchemaField(
-			false,
-			JSONUtil.getValueAsJSONArray(
-				_getJSONObject(mutationFieldsJSONArray, "testPath_v1_0"),
-				"JSONObject/type", "JSONArray/fields"),
-			true, "createTestDTO");
+			true, mutationFieldsJSONArray, true, "createTestDTO");
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java:456]: This assertion is duplicated. See the line 451

File: modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java [hunk 2]
```diff
@@ -458,48 +453,25 @@ public void testSchema() throws Exception {
...
 
 		// Query fields
 
-		JSONArray queryFieldsJSONArray = JSONUtil.getValueAsJSONArray(
+		JSONArray typeFieldsJSONArray = JSONUtil.getValueAsJSONArray(
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java:460]: You removed the assertion over the `queryType` graphQL field. The type has to be done for both `mutationType` and `queryType`

File: modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java [hunk 3]
```diff
@@ -458,48 +453,25 @@ public void testSchema() throws Exception {
...
 			_invoke(
 				new GraphQLField(
-					"__schema",
+					"__type(name: \"TestPath_v1_0\")",
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/servlet/test/GraphQLServletTest.java:463]: I miss an explanation in the commit message why it is necessary to update the tests

**Review by ccorreagg — CHANGES_REQUESTED** [review]
Please see my comments

**Comment by GaborKomaromi** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**